### PR TITLE
[1LP][RFR] Check for badness rework

### PR DESCRIPTION
**Pull request by psav**
We are running things in the browser without first checking for modal dialog, I decided it may be better to put all the precheck stuff inside the try block. This is a PoC really, and I think check_for_badness really needs to be redesigned.

### CODE REVIEW

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by mshriver** [comment]
I believe this is the root cause of a large number of failures in the 17.26.0-rc test run. Once an UnexpectedAlertPresentException is encountered in a browser session, subsequent tests on that session will fail to navigate.  I think we should merge this for the 17.26.0 release.

File: cfme/utils/appliance/implementations/ui.py [hunk 1]
```diff
@@ -287,6 +276,20 @@ def check_for_badness(self, fn, _tries, nav_args, *args, **kwargs):
             self.go(_tries, *args, **go_kwargs)
             # If there is a rails error past this point, something is really awful
 
+    def check_for_badness(self, fn, _tries, nav_args, *args, **kwargs):
+        if getattr(fn, '_can_skip_badness_test', False):
```
Comment (jamesooden) [inline cfme/utils/appliance/implementations/ui.py]: **QUESTION:** What is fn?

Comment (mshriver) [reply cfme/utils/appliance/implementations/ui.py]: fn is 'function', check_for_badness is a wrapper for step/am_i_here and other NavigateStep methods.

Comment (psav) [reply cfme/utils/appliance/implementations/ui.py]: It's all a little tiny, insignificant;y small bit of a mess :)

**Comment by mshriver** [comment]
PRT looks good, local testing looks good - I verified that with an unexpected modal/dialog present, navigation dismisses it.

**Pull request merged**